src/App.tsx

```diff
@@ -1,20 +1,58 @@
-import React from 'react';
+import React, { ChangeEvent, useCallback, useEffect, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import debounce from 'lodash.debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedName, setSelectedName] =
+    useState<string>('No selected person');
+  const [personList, setPersonList] = useState(peopleFromServer);
+  const [inputName, setInputName] = useState('');
+  const [appliedInput, setAppliedInput] = useState('');
+  const [isInputFocus, setIsInputFocus] = useState(true);
+
+  const applyInput = useCallback((value: string) => {
+    debounce(() => setAppliedInput(value), 300)();
+  }, []);
+
+  useEffect(() => {
+    if (inputName.trim() === '') {
+      setPersonList(peopleFromServer);
+    } else {
+      setPersonList(
+        peopleFromServer.filter(person =>
+          person.name.toLowerCase().includes(appliedInput.toLowerCase()),
+        ),
+      );
+    }
+  }, [appliedInput]);
+
+  function handlePersonClick(person: Person) {
+    setInputName(person.name);
+    setSelectedName(`${person.name} (${person.born} - ${person.died})`);
+    setIsInputFocus(false);
+  }
+
+  function handleChangeInput(event: ChangeEvent<HTMLInputElement>) {
+    setInputName(event.target.value);
+    applyInput(event.target.value);
+    setSelectedName('No selected person');
+  }
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {selectedName}
         </h1>
 
-        <div className="dropdown is-active">
+        <div className={`dropdown ${isInputFocus ? 'is-active' : ''} `}>
           <div className="dropdown-trigger">
             <input
+              value={inputName}
+              onClick={() => setIsInputFocus(true)}
```

Comment:
You should use [`onFocus`](https://developer.mozilla.org/en-US/docs/Web/API/Element/focus_event) and [`onBlur`](https://developer.mozilla.org/en-US/docs/Web/API/Element/blur_event) to manage input focus

Reply:
Hello, Edu! I had tried this initially and noticed that there was a conflict between OnBlur and OnClick. Anyway, I found a way to make the two work together using setTimeout. I believe that with UseEffect it would also work. Thanks!